packages/components/src/components/textarea/textarea.ts

```diff
@@ -247,93 +249,150 @@ export default class SdTextarea extends SolidElement implements SolidFormControl
+    };
+
+    // States
+    const hasLabel = this.label ? true : !!slots['label'];
+    const hasHelpText = this.helpText ? true : !!slots['helpText'];
+    const hasValidationAttr = this.required || !!this.minlength || !!this.maxlength;
+    const isInvalid = hasValidationAttr && !this.checkValidity();
+    const isValid = hasValidationAttr && this.checkValidity();
+    // Hierarchy of input states:
+    const inputState = this.disabled
+      ? 'disabled'
+      : this.readonly
+        ? 'readonly'
+        : this.hasFocus && isInvalid
+          ? 'activeInvalid'
+          : this.hasFocus && isValid
+            ? 'activeValid'
+            : this.hasFocus
+              ? 'active'
+              : isInvalid
+                ? 'invalid'
+                : isValid
+                  ? 'valid'
+                  : 'default';
+
+    // Conditional Styles
+    const textSize = this.size === 'sm' ? 'text-sm' : 'text-base';
+    const textColor = {
+      disabled: 'text-neutral-500',
+      readonly: 'text-black',
+      activeInvalid: 'text-error',
+      activeValid: 'text-success',
+      active: 'text-black',
+      invalid: 'text-error',
+      valid: 'text-success',
+      default: 'text-black'
+    }[inputState];
+    const borderColor = {
+      disabled: 'border-neutral-500',
+      readonly: 'border-neutral-800',
+      activeInvalid: 'border-error border-2',
+      activeValid: 'border-success border-2',
+      active: 'border-primary border-2',
+      invalid: 'border-error',
+      valid: 'border-success',
+      default: 'border-neutral-800'
+    }[inputState];
 
     return html`
       <div
         part="form-control"
-        class=${classMap({
-      'form-control': true,
-      'form-control--small': this.size === 'small',
-      'form-control--medium': this.size === 'medium',
-      'form-control--large': this.size === 'large',
-      'form-control--has-label': hasLabel,
-      'form-control--has-help-text': hasHelpText
-    })}
+        class=${cx(
+          'form-control text-left',
+          hasLabel && 'form-control--has-label',
+          hasHelpText && 'form-control--has-help-text',
+          this.disabled && 'pointer-events-none cursor-not-allowed'
+        )}
       >
         <label
           part="form-control-label"
-          class="form-control__label"
+          class=${cx('form-control-label mb-2', hasLabel ? 'inline-block' : 'hidden', textSize)}
           for="input"
           aria-hidden=${hasLabel ? 'false' : 'true'}
         >
-          <slot name="label">${this.label}</slot>
+          <slot name="label">${this.label}${this.required ? '*' : ''}</slot>
         </label>
 
-        <div part="form-control-input" class="form-control-input">
+        <div part="form-control-input" class="form-control-input relative w-full">
+          <div
+            part="border"
+            class=${cx('absolute w-full h-full pointer-events-none border rounded-default', borderColor)}
```

Comment:
Is there a specific reason why you've resolved 'borderColor' as a constant rather than directly resolving it within the class as follows?

```
{
  disabled: 'border-neutral-500',
  readonly: 'border-neutral-800',
  activeInvalid: 'border-error border-2',
  activeValid: 'border-success border-2',
  active: 'border-primary border-2',
  invalid: 'border-error',
  valid: 'border-success',
  default: 'border-neutral-800'
}[inputState];
```